src/OrbitGl/App.cpp

```diff
@@ -2829,21 +2829,49 @@ void OrbitApp::SelectCallstackEvents(const std::vector<CallstackEvent>& selected
   GetMutableCaptureData().set_selection_callstack_data(std::move(selection_callstack_data));
 
   // Generate selection report.
-  bool generate_summary = origin_is_multiple_threads;
   PostProcessedSamplingData selection_post_processed_sampling_data =
       orbit_client_model::CreatePostProcessedSamplingData(
           GetCaptureData().selection_callstack_data(), GetCaptureData(), *module_manager_,
-          generate_summary);
+          /*generate_summary*/ origin_is_multiple_threads);
   GetMutableCaptureData().set_selection_post_processed_sampling_data(
       std::move(selection_post_processed_sampling_data));
+}
 
+void OrbitApp::SelectCallstackEvents(const std::vector<CallstackEvent>& selected_callstack_events,
+                                     bool origin_is_multiple_threads) {
+  ClearInspection();
+  SetCaptureDataSelectionFields(selected_callstack_events, origin_is_multiple_threads);
   SetSelectionTopDownView(GetCaptureData().selection_post_processed_sampling_data(),
                           GetCaptureData());
   SetSelectionBottomUpView(GetCaptureData().selection_post_processed_sampling_data(),
                            GetCaptureData());
-
   SetSelectionReport(&GetCaptureData().selection_callstack_data(),
-                     &GetCaptureData().selection_post_processed_sampling_data(), generate_summary);
+                     &GetCaptureData().selection_post_processed_sampling_data(),
+                     /*has_summary*/ origin_is_multiple_threads);
+}
+
+void OrbitApp::InspectCallstackEvents(const std::vector<CallstackEvent>& selected_callstack_events,
+                                      bool origin_is_multiple_threads) {
+  if (absl::GetFlag(FLAGS_time_range_selection)) {
+    SetCaptureDataSelectionFields(selected_callstack_events, origin_is_multiple_threads);
```

Comment:
I am not sure of my understanding of what's happening here with this call to `SetCaptureDataSelectionFields`. The selection is overwritten by the inspection, although the "(selection)" tabs are left alone for the time being?

Reply:
Exactly. The reason for this is so that the bright green highlighting of the callstacks on the capture tab is done correctly (FireRefreshCallbacks makes the capture tab refresh and it uses these vectors to know what to highlight). LMK if there's a better way to be doing this though.

Reply:
> LMK if there's a better way to be doing this though.

I doubt it.